framework/include/mesh/FaceInfo.h

```diff
@@ -62,32 +63,36 @@ class FaceInfo
   /// Returns the unit normal vector for the face oriented outward from the face's elem element.
   const Point & normal() const { return _normal; }
 
-  /// Returns true if this face resides on the mesh boundary.
-  bool isBoundary() const { return (_neighbor == nullptr); }
-
   /// Returns the coordinates of the face centroid.
   const Point & faceCentroid() const { return _face_centroid; }
 
   ///@{
-  /// Returns the coordinates of the approximate face centroid
-  /// (intersection of the face and the line between the cell centroids)
-  /// in case of skewed meshes.
-  const Point & rIntersection() const { return _r_intersection; }
+  /// Returns the skewness-correction vector (vector between the approximate and real face
+  /// centroids).
+  const Point skewnessCorrectionVector() const
+  {
+    Point r_intersection =
+        _elem_info->centroid() +
+        (((_face_centroid - _elem_info->centroid()) * _normal) / (_e_cn * _normal)) * _e_cn;
+
+    return _face_centroid - r_intersection;
+  }
   ///@}
 
   ///@{
   /// Returns the elem and neighbor elements adjacent to the face.
   /// If a face is on a mesh boundary, the neighborPtr
   /// will return nullptr - the elem will never be null.
-  const Elem & elem() const { return *_elem; }
-  const Elem * neighborPtr() const { return _neighbor; }
+  const Elem & elem() const { return *(_elem_info->elem()); }
+  const Elem * elemPtr() const { return _elem_info->elem(); }
   const Elem & neighbor() const
   {
-    if (!_neighbor)
+    if (!_neighbor_info->elem())
       mooseError("FaceInfo object 'const Elem & neighbor()' is called but neighbor element pointer "
                  "is null. This occurs for faces at the domain boundary");
-    return *_neighbor;
+    return *(_neighbor_info->elem());
   }
```

Comment:
Any multi-line method definitions, let's move out of the class definition. You can put them as `inline` in the header if you want to keep them inlineable

Reply:
Sounds good, I think I understand, will pay attention to this next time!